wordle_solver/wordle_solver.py

```diff
@@ -0,0 +1,66 @@
+from enum import Enum
+from string import ascii_lowercase
+from itertools import groupby
+
+from .word_initializers.english_words_initilalizer import EnglishWords
+
+WORDS_LENGTH = 5
+
+
+class Languages(Enum):
+    ENGLISH = EnglishWords
+
+
+class WordleSolver:
+    def __init__(self, words_length: int, language: str):
+        self.words_length: int = words_length
+        self.all_words: set[str] = Languages[language].value(self.words_length).get_words()
+        self.keys: dict[str, set[int]] = dict()
+        self.must_haves: set[str] = set()
+        self.reset()
+
+    def reset(self):
+        self.keys = {letter: {i for i in range(self.words_length)} for letter in ascii_lowercase}
+        self.must_haves = set()  # each character points to the number of times it could occur.
+
+    def insert_green_letter(self, letter: str, index: int):
+        self.must_haves.add(letter)
+        self.keys[letter] = {index}
+
+    def insert_yellow_letter(self, letter: str, index: int):
+        self.must_haves.add(letter)
+        self.keys[letter].discard(index)
+
+    def insert_black_letter(self, letter: str, index: int):
+        self.keys[letter] = set()
+
+    def valid_word(self, word: str) -> bool:
+        return all([index in self.keys[letter] for index, letter in enumerate(word)]) and \
+               all([letter in word for letter in self.must_haves])
+
+    def guess_word(self, word: str, colors: str) -> bool:
```

Comment:
After certain guess sequences, there's no possible words left.
Maybe print a message if this happens? Or return a second value?

Reply:
Not yet, since:
1) The algorithm does not support double letters
2) We might encounter words that are not in our vocabulary but used by other wordle-esque games.
But it could and even should be mentioned in case your CLI will have an option of printing possible solutions.

Reply:
Had no idea there could be words out of our vocabulary..! Okay, this changes a lot.
Gotcha. I'll find an appropriate place to add a message to.